### DISCUSSION TITLE
How neural networks parameters initializer works ?

### DISCUSSION THREAD

**Discussion by jrabary** [ml-explore/mlx#417] [General]
It looks like currently layer such as Linear doesn't have optional bias and kernel initializer. What workaround we have to use other initializers than uniform ? Did you plan to support it in the future ?

**Comment by awni** [comment]
The easiest thing would be to do `mylinear.weight = <my_custom_init_fn>`. 

If you want to change many weights you can use [`Module.apply`](https://ml-explore.github.io/mlx/build/html/python/nn/_autosummary/mlx.nn.Module.apply.html#mlx.nn.Module.apply) with a filter to choose the weights.

We may add more initializers in the future...

**Comment by jrabary** [comment]
That was easy. Thanks a lot.
